test/integration/upload.spec.ts

```diff
@@ -0,0 +1,205 @@
+    // Create PubSub room to listen on events
+    roomPinner = await createLibp2pRoom(libp2p, testApp.providerAddress)
+  })
+  after(async () => {
+    await testApp.stop()
+  })
+
+  it('should create upload folder', () => {
+    const uploadFolderExist = fs.existsSync(path.resolve(process.cwd(), UPLOAD_FOLDER))
+    expect(uploadFolderExist).to.be.eql(true)
+  })
+  it('should init provider and libp2p', () => {
+    const libp2p = testApp.app?.app.get('libp2p')
+    const storageProvider = testApp.app?.app.get('storage')
+    expect(libp2p).to.be.instanceOf(Libp2p)
+    expect(storageProvider).to.be.instanceOf(ProviderManager)
+  })
+  describe('GC', () => {
+    let gcInterval: string
+    let jobTtl: string
+    let gc: { stop: () => void }
+    before(() => {
+      gcInterval = config.get<string>('gc.interval')
+      jobTtl = config.get<string>('gc.jobTtl')
+      // @ts-ignore
+      config.gc.interval = '5s'
+      // @ts-ignore
+      config.gc.jobTtl = '1s'
+      gc = jobsGC(testApp.app!.app)
+    })
+    after(() => {
+      // @ts-ignore
+      config.gc.interval = gcInterval
+      // @ts-ignore
+      config.gc.jobTtl = jobTtl
+      gc.stop()
+    })
+    afterEach(async () => {
+      await UploadJob.destroy({ where: {} })
+    })
+    it('should remove and unpin expired files', async () => {
+      const file1Response = await upload(testApp.providerAddress, 'testAccount', testApp.peerId?.id as string, './test/integration/files/testFile.txt')
+      const file2Response = await upload(testApp.providerAddress, 'testAccount', testApp.peerId?.id as string, './test/integration/files/testFile2.txt')
+      const file3Response = await upload(testApp.providerAddress, 'testAccount', testApp.peerId?.id as string, './test/integration/files/testFile3.txt')
+
+      expect(rooms.size).to.be.eql(1)
+
+      expect(file1Response.message).to.be.eql('File uploaded')
+      expect(file1Response.fileHash).to.be.not.null()
+      const job1 = await UploadJob.findOne({ where: { fileHash: `/ipfs/${file1Response.fileHash}` } })
+      expect(await isPinned(ipfs, new CID(file1Response.fileHash))).to.be.true()
+      expect(job1).to.be.instanceOf(UploadJob)
+
+      expect(file2Response.message).to.be.eql('File uploaded')
+      expect(file2Response.fileHash).to.be.not.null()
+      const job2 = await UploadJob.findOne({ where: { fileHash: `/ipfs/${file2Response.fileHash}` } })
+      expect(await isPinned(ipfs, new CID(file2Response.fileHash))).to.be.true()
+      expect(job2).to.be.instanceOf(UploadJob)
+
+      expect(file3Response.message).to.be.eql('File uploaded')
+      expect(file3Response.fileHash).to.be.not.null()
+      const job3 = await UploadJob.findOne({ where: { fileHash: `/ipfs/${file3Response.fileHash}` } })
+      expect(await isPinned(ipfs, new CID(file3Response.fileHash))).to.be.true()
+      expect(job3).to.be.instanceOf(UploadJob)
+
+      await sleep(7000)
```

Comment:
Maybe could we make it faster? We should avoid long-running tests whenever we can. I think you can use milliseconds for the GC settings right?

Reply:
Done